compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -17752,30 +17752,34 @@ TR::Node *fmaxminSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
    TR::Node * secondChild = node->getSecondChild();
    bool isBothConst = firstChild->getOpCode().isLoadConst() && secondChild->getOpCode().isLoadConst();
    float fmin = 0, fmax = 0;
-   bool maxOpcode = node->getOpCodeValue() == TR::fmax;
+   bool isMaxOpcode = node->getOpCodeValue() == TR::fmax;
 
-   if (isBothConst)
+   if (!isBothConst) return node;
+
+   float first = firstChild->getFloat();
+   float second = secondChild->getFloat();
+
+   uint32_t firstBits = firstChild->getFloatBits();
+   uint32_t secondBits = secondChild->getFloatBits();
+
+   if (first != first) // this is true only for NaNs
```

Comment:
I understand this comparison that you did, but actually prefer what we had originally. That is more readable and checks bit pattern as well.

Reply:
would it be better to change the implementation of isNaNFloat and isDoubleFloat to use `value != value`? Or is there some other reason why we check the bit pattern? NaN != NaN is true even if the bit patterns of each NaN are different

Reply:
Yes, as I said, I understand `value != value` will be true for NaN, I think the explicit check is the preferred way.